src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -84,68 +82,92 @@ public async Task<bool> PushMetadataAsync(CancellationToken cancellationToken)
                 }
                 else
                 {
-                    (List<BuildData> buildsManifestMetadata,
-                     List<SigningInformation> signingInformation,
-                     ManifestBuildData manifestBuildData) = GetBuildManifestsMetadata(ManifestsPath, cancellationToken);
+                    //get the list of manifests
+                    List<Manifest> parsedManifest = GetParsedManifest(ManifestsPath, cancellationToken);
 
-                    if (buildsManifestMetadata.Count == 0)
+                    if (parsedManifest.Count == 0)
                     {
-                        Log.LogError($"No build manifests found matching the search pattern {SearchPattern} in {ManifestsPath}");
+                        Log.LogError(
+                            $"No manifests found matching the search pattern {SearchPattern} in {ManifestsPath}");
                         return !Log.HasLoggedErrors;
                     }
 
-                    BuildData finalBuild = MergeBuildManifests(buildsManifestMetadata);
-                    IMaestroApi client = ApiFactory.GetAuthenticated(MaestroApiEndpoint, BuildAssetRegistryToken);
-
-                    var deps = await GetBuildDependenciesAsync(client, cancellationToken);
-                    Log.LogMessage(MessageImportance.High, "Calculated Dependencies:");
-                    foreach (var dep in deps)
+                    Manifest manifest;
+                    //check if the manifest have any duplicate packages and blobs
+                    if (parsedManifest.Count > 1)
                     {
-                        Log.LogMessage(MessageImportance.High, $"    {dep.BuildId}, IsProduct: {dep.IsProduct}");
+                        manifest = CheckIfManifestCanBeMerged(parsedManifest);
```

Comment:
Nit: This should probably be named something like `MergeManifests`, since that's actually what it's doing.

Reply:
done